solana/pyth2wormhole/client/src/main.rs

```diff
@@ -190,301 +225,276 @@ fn handle_attest(
-                                    conf_timeout.as_secs(),
-                                    conf_timeout.subsec_millis(),
-				    attempt_no, n_retries,
-				    msg
-                                );
-
-                                tx_confirm_failures += 1;
-
-                                if attempt_no < &n_retries {
-                                    state.set_status(Sending {
-                                        attempt_no: attempt_no + 1,
-                                    });
-                                } else {
-                                    error!(
-                                        "Batch {}/{} (group {:?}) tx confirm: All {} attempts failed",
-                                        state.group_name,
-                                        batch_no,
-                                        batch_count,
-                                        n_retries + 1
-                                    );
-                                    state.set_status(FailedConfirm { last_err: e });
-                                }
-                            }
-                        }
-                    }
-                }
-                Success { .. } | FailedSend { .. } | FailedConfirm { .. } => {
-                    // We only try to re-schedule under --daemon
-                    if daemon {
-                        if let Some(reason) = state.should_resend(rpc_client) {
-                            info!(
-                                "Batch {}/{} (group {:?}): resending (reason: {})",
-                                batch_no, batch_count, state.group_name, reason,
-                            );
-                            state.set_status(Sending { attempt_no: 1 });
-                        } else {
-                            let elapsed = state.get_status_changed_at().elapsed();
-                            trace!(
-                                "Batch {}/{} (group {:?}): waiting ({}.{}s elapsed)",
-                                batch_no,
-                                batch_count,
-                                state.group_name,
-                                elapsed.as_secs(),
-                                elapsed.subsec_millis(),
-                            )
-                        }
-                    } else {
-                        // Track the finished batches outside daemon mode
-                        finished_count += 1;
-
-                        // No RPC requests are made on terminal states outside daemon mode, skip sleep
-                        continue;
-                    }
-                }
+    // With daemon mode off, the sched jobs return from the
+    // join_all. We filter out errors and report them
+    let errors: Vec<_> = results
+        .iter()
+        .filter_map(|r| {
+            if let Err(e) = r {
+                Some(e.to_string())
+            } else {
+                None
             }
```

Comment:
You could use `Result::err` here: https://doc.rust-lang.org/std/result/enum.Result.html#method.err